test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs

```diff
@@ -21,6 +21,17 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
         public HttpClient Client { get; }
 
         [Fact]
+        public async Task RazorPage_TempDataFilters_DontSurviveBetweenPages()
+        {
+            var response = await Client.GetAsync("http://localhost/TempData/TempDataPageModelProperty");
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            //If the filter from above was re-used in the below request the request would fail due to type cast.
+            response = await Client.GetAsync("http://localhost/HandlerTestPage");
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

Comment:
It's testing that Factories (and therefore TempDataProperties) are per-page and don't "leak". This test failed before I added the new `ConfigureFilter` overload and used that. I tried to describe that in the comment above, but I'll take another stab at clarifying.